src/package-request.js

```diff
@@ -52,6 +53,7 @@ export default class PackageRequest {
   config: Config;
   registry: ResolverRegistryNames;
   optional: boolean;
+  hint: ?string;
```

Comment:
Would you mind changing this to an enum both to avoid unnecessary string usage all over the app and to avoid potential typing errors.

My suggestion is to add something like the following into constants.js and use that:
```
const REQUEST_HINTS = Object.freeze({
    dev: Symbol('REQUEST_HINT_DEV'),
    optional: Symbol('REQUEST_HINT_OPTIONAL'),
});
```